### DISCUSSION TITLE
How renovate updates pubspec.yaml with dependencies on private git repositories

### DISCUSSION THREAD

**Discussion by fossamagna** [renovatebot/renovate#23703] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

## results

I have a flutter plugin (ex: pluginA) on private repo in github.com, and I have a flutter app dependency to pluginA.
The following error occurs when updating pubspec.lock.
```
Command failed: docker run --rm --name=renovate_a_sidecar --label=renovate_a_child --memory=3584m -v "/tmp/worker/2aaa13/8d8e29/repos/github/sample_org/sample_app":"/tmp/worker/2aaa13/8d8e29/repos/github/sample_org/sample_app" -v "/tmp/worker/2aaa13/8d8e29/cache":"/tmp/worker/2aaa13/8d8e29/cache" -e CONTAINERBASE_CACHE_DIR -w "/tmp/worker/2aaa13/8d8e29/repos/github/sample_org/sample_app" ghcr.io/containerbase/sidecar:9.8.0 bash -l -c "install-tool flutter 3.10.6 && flutter pub upgrade hooks_riverpod"
/usr/local/bin/docker: line 4: .: filename argument required
.: usage: . filename [arguments]
Downloading Linux x64 Dart SDK from Flutter engine cdbeda788a293fa29665dc3fa3d6e63bd221cb0d...
Building flutter tool...
Git error. Command: `git clone --mirror git@github.com:sample_org/flutter_reactive_ble.git /home/ubuntu/.pub-cache/git/cache/flutter_reactive_ble-bfacb3d5dd0137eac59da4e472003345ec52d283`
stdout: 
stderr: Cloning into bare repository '/home/ubuntu/.pub-cache/git/cache/flutter_reactive_ble-bfacb3d5dd0137eac59da4e472003345ec52d283'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
exit code: 128
```

## expected behavior

- Update pubspec.yaml without error.
- Do not commit secret values to the git repository (ex: GH token) 

### What you tried so far.

The following steps to worked:

1. Use https schema for URL of git private repo in pubspec.yaml.
2. add personal access token to URL of git private repo. (ex: `https://oauth2:github_pat_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx@github.com/sample_org/sample_pluginA.git`)

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
the git error should be gone, please try again

**Reply by fossamagna** [reply]
@viceice 
Sorry for the delay! The git error be gone.
Thank you!
